utils/fetch-metadata/src/url.rs

```diff
@@ -183,9 +171,19 @@ async fn fetch_metadata(client: impl ClientT, version: MetadataVersion) -> Resul
         Ok(metadata.0)
     }
 
-    // Fetch using the new interface, falling back to trying old one if there's an error.
-    match fetch_inner(&client, version).await {
-        Ok(s) => Ok(s),
-        Err(_) => fetch_inner_legacy(&client, version).await,
+    match fetch_available_versions(&client).await {
+        Ok(supported_versions) => {
+            fetch_inner(&client, version, supported_versions).await
+        },
+        Err(e) => {
+            // The "new" interface failed. if the user is asking for V14 or the "latest"
+            // metadata then try the legacy interface instead. Else, just return the
+            // reason for failure.
+            if matches!(version, MetadataVersion::Version(14) | MetadataVersion::Latest) {
+                fetch_inner_legacy(&client).await
+            } else {
+                Err(e)
+            }
+        }
     }
```

Comment:
I tidied this up because previously it was returning only the error from `fetch_inner_legacy` (ie "The node can only return version 14 metadata using the legacy API..") which wasn't very helpful!